src/routes/(app)/profile/modals/PwChangeModal.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+  import Modal from "@/lib/Modal.svelte";
+  import type { ChangePasswordForm } from "@/types";
+  import { changeUserPassword } from "$lib/util/api";
+
+  export let userName: string | undefined;
+  export let onClose: () => void;
+  export let changepswd: ChangePasswordForm = {
+    currentPassword: "",
+    newPassword: "",
+    reEnteredNewPassword: ""
+  };
+
+  let error: string;
+  let errs: string[] = [];
+
+  function checkForm() {
+    errs = [];
+    console.log("Change Password Form - Status: Checking if any form inputs are missing");
+    if (!changepswd.currentPassword) {
+      errs.push("Current Password");
+    }
+    if (!changepswd.newPassword) {
+      errs.push("New Password");
+    }
+    if (!changepswd.reEnteredNewPassword) {
+      errs.push("Re-Entered New Password");
+    }
+    if (errs.length > 0) {
+      error = `Missing required params: ${errs.join(", ")}`;
+      console.log(
+        `Change Password Form - Error: following form inputs are missing:\n${errs.join("\n")}`
+      );
+    } else {
+      console.log("Change Password Form - Status: All form inputs are present");
+      checkFormPasswordsMatch();
+    }
+  }
+
+  function checkFormPasswordsMatch() {
+    console.log(
+      "Change Password Form - Status: Checking if new password and re-entered new password match"
+    );
+    if (changepswd.newPassword !== changepswd.reEnteredNewPassword) {
+      error = "New password and re-entered new password do not match";
+      console.log(
+        "Change Password Form - Error: New password and re-entered new password do not match"
+      );
+    } else {
+      console.log("Change Password Form - Status: New password and re-entered new password match");
+      error = "";
+    }
+  }
+
+  function handleSubmit(ev: SubmitEvent) {
+    checkForm();
+    if (!error) {
+      console.log(
+        "Change Password Form - Status: Form inputs on the frontend are valid - sending request to change password to the server"
+      );
+      const fd = new FormData(ev.target! as HTMLFormElement);
+      changepswd.currentPassword = fd.get("current-password") as string;
+      changepswd.newPassword = fd.get("new-password") as string;
+      changeUserPassword(changepswd.currentPassword, changepswd.newPassword, () => {
```

Comment:
Related to the other comment, it could then be used like so:

```
changeUserPassword(changepswd.currentPassword, changepswd.newPassword, (err) => {
  if (err) {
    error = err;
    return;
  }
  console.log("Change Password Form - Status: Password changed successfully");
  onClose();
});
```